### PR TITLE
small fixes to better support 'show more events'

### PR DESCRIPTION
**Pull request by darsney**
and associated perf improvements for very large grids

### CODE REVIEW

File: src/UXClient/Models/EventsTableData.ts [hunk 1]
```diff
@@ -11,19 +11,25 @@ class EventsTableData {
...
     private offsetName = null;
     private maxVisibleToStart = 10;
+    private offsetNameCache = {};
 
 	constructor(){
         
     }
 
     private createOffsetName (offset) {
+        if(offset in this.offsetNameCache){
```
Comment (ccrowley96) [inline src/UXClient/Models/EventsTableData.ts:21]: woo woo

File: src/UXClient/Models/TimeSeriesEvent.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ class TimeSeriesEvent {
             let utcOffsetDate = Utils.offsetUTC(new Date(Date.parse(rawEvent['timestamp ($ts)'].split("Z").join(""))));
             rawEvent[offsetName + "_" + type] = {
                 name: offsetName,
-                value: Utils.timeFormat(true, true, offset, true, null, null, locale)(utcOffsetDate),
+                value: () => Utils.timeFormat(true, true, offset, true, null, null, locale)(utcOffsetDate),
```
Comment (ccrowley96) [inline src/UXClient/Models/TimeSeriesEvent.ts:14]: This function call is wild: true, true, **offset**, true, null, null, locale

File: src/UXClient/Models/EventsTableData.ts (hunk 1 above)
Comment (niusoff) [reply src/UXClient/Models/EventsTableData.ts:21]: **offset**

File: src/UXClient/Models/EventsTableData.ts [hunk 3]
```diff
@@ -11,19 +11,25 @@ class EventsTableData {
...
+            return this.offsetNameCache[offset];
+        }
         var offsetSubstring = "";
         if ((typeof offset) === 'string') {
             offsetSubstring = Utils.convertTimezoneToLabel(offset);
         } else {
             offsetSubstring = Utils.formatOffsetMinutes(offset);
         }
-        return "timestamp " + offsetSubstring;
+        let offsetName = "timestamp " + offsetSubstring;
```
Comment (niusoff) [inline src/UXClient/Models/EventsTableData.ts:30]: out of the scope of this code review, but should we consider loc-ing the word "timestamp" in the future?

File: src/UXClient/Models/TimeSeriesEvent.ts (hunk 2 above)
Comment (darsney) [reply src/UXClient/Models/TimeSeriesEvent.ts:14]: yeahhh in the future i think we should move to making function calls that have a lot of params like this to take a single object that is strongly typed

File: src/UXClient/Models/EventsTableData.ts (hunk 3 above)
Comment (darsney) [reply src/UXClient/Models/EventsTableData.ts:30]: ya def

**Pull request merged**